src/scanner.ts

```diff
@@ -215,10 +218,29 @@ const affectedPackageNames = new Set(
  * Fast membership check for whether a package name appears in the compromised
  * master package list.
  * @param packageName The dependency name to check.
+ * @param version Optional specific version to check (defaults to '*').
  * @returns true if the package is flagged as affected.
  */
-export function isAffected(packageName: string): boolean {
-	return affectedPackageNames.has(packageName);
+export function isAffected(packageName: string, version: string = '*'): boolean {
+	if (affectedPackageNames.has(packageName)) {
+		const pkg = masterPackages.packages.find((p) => p.name === packageName);
+		if (!pkg) return false;
+
+		if (version === '*' || pkg.affectedVersions.includes('*')) {
+			return true;
+		}
+		if (pkg.affectedVersions.includes(version)) {
+			return true;
+		}
+		try {
+			const semverVersion = new SemVer(version, {loose: true});
+			return pkg.affectedVersions.some(range => satisfies(semverVersion, range));
+		} catch (e) {
+			// Invalid semver version, probably because version is itself a range from package.lock
+			return pkg.affectedVersions.some(range => intersects(version, range, {loose: true}));
```

Comment:
`SemVer` constructor will throw an error if the supplied version is not a single specific version string, but e.g. a version range pattern (`^1.2.3` or `>=1.2.0` etc.). We assume here that if the version of the package given is not a proper (loose) version string, it is coming from the `package.json` file and a version range specifier, hence we attempt an `intersect` match.